src/Kernels.h

```diff
@@ -1,16 +1,122 @@
 #ifndef COVIDSIM_KERNELS_H_INCLUDED_
 #define COVIDSIM_KERNELS_H_INCLUDED_
 
-extern double *nKernel, *nKernelHR;
-
-void InitKernel(double);
-double ExpKernel(double);
-double PowerKernel(double);
-double PowerKernelB(double);
-double PowerKernelUS(double);
-double PowerExpKernel(double);
-double GaussianKernel(double);
-double StepKernel(double);
-double numKernel(double);
+#include <vector>
+
+struct Cell;
+
+namespace CovidSim
+{
+  /// \todo TBD1 is specific to this file and is to be changed to something meaningful during a later refactoring
+  namespace TBD1
+  {
+    /// A probability distribution
+    struct KernelStruct
+    {
+      /// Which distribution to use
+      int type_;
+
+      /// Distribution parameter
+      double shape_;
+
+      /// Representative distance
+      double scale_;
+
+      /// Distribution parameter
+      double p3_;
+
+      /// Distribution parameter
+      double p4_;
+
+      /// \param r2 The distance squared
+      /// \return Probability
+      double exponential(double r2) const;
+
+      /// \param r2 The distance squared
+      /// \return Probability
+      double power(double r2) const;
+
+      /// \param r2 The distance squared
+      /// \return Probability
+      double power_b(double r2) const;
+
+      /// \param r2 The distance squared
+      /// \return Probability
+      double power_us(double r2) const;
+
+      /// \param r2 The distance squared
+      /// \return Probability
+      double power_exp(double r2) const;
+
+      /// Gaussian distribution a.k.a. normal distribution and bell curve
+      /// \param r2 The distance squared
+      /// \return Probability
+      double gaussian(double r2) const;
+
+      /// Step function
+      /// \param r2 The distance squared
+      /// \return Probability
+      double step(double r2) const;
+    };
+
+    /// \brief To speed up calculation of kernel values we provide a couple of lookup
+    /// tables.
+    ///
+    /// lookup_ is a table of lookups. lookup_[0] is the kernel
+    /// value at a distance of 0, and lookup_.back() is the kernel value at the
+    /// largest possible distance (diagonal across the bounding box).
```

Comment:
`(diagonal across the bounding box)` is incorrect with spherical co-ordinates.  See https://github.com/mrc-ide/covid-sim/blob/master/src/SetupModel.cpp#L201

Reply:
Bit obvious I suppose - the bounding box could straddle the equator and we do live on an oblate spheroid.